py/desispec/preproc.py

```diff
@@ -609,7 +612,17 @@ def preproc(rawimage, header, primary_header, bias=True, dark=True, pixflat=True
         else :
             rdnoise_message = 'ADUs (gain not applied)'
             gain_message    = 'gain not applied to image'
-        header['OBSRDN'+amp] = (median_rdnoise,rdnoise_message)
+
+        if 'preproc' in desispec_params and len(rawimage)>4000: # There are 1100*900 b0 image in the test script
+            header['OVSRDN'+amp] = (median_rdnoise,rdnoise_message)
+            ind=desispec_params['preproc']['rdnoise']['PARAMS']['OBSRDN'+amp+'_'+camera.upper()[0]]
+            ind_sec = parse_sec_keyword(ind)
+            d=rawimage[ind_sec]
+            d = d[~np.isnan(d)].ravel()
+            header['OBSRDN'+amp]=(gain*(np.percentile(d,84)-np.percentile(d,16))/2.,rdnoise_message)
```

Comment:
Please use the *same* statistical method for calculating the readnoise in the overscan vs. the data region.  Since the data are quantized, the method here (difference of percentiles) will result in quantized readnoise measurements.  Please see how the readnoise is calculated in the `_overscan()` function (std of sigma-clipped pixels, with a correction for the bias from the sigma-clipping), pull that out into a separate function that both can call, and then use that function for calculating the readnoise in both cases.

Additionally, that function should accept a mask so that known bad pixels can be pre-flagged and ignored without potentially confusing the sigma-clipping.

Ideally cosmics would also be ignored (i.e. part of the mask), but that creates a chicken-and-the-egg problem: cosmic finding uses the estimated variance, but the variance depends upon the readnoise, but we want to mask cosmics when calculating readnoise.  Maybe the sigma clipping will be sufficient for that, but we'll have to watch out for outlier readnoise measurements and perhaps adjust (e.g. using a fast approximate cosmic finder first, or using the overscan readnoise for the cosmics, and then updating with the final data-region readnoise).